pkg/kubelet/rkt/image.go

```diff
@@ -66,30 +66,36 @@ func (r *Runtime) PullImage(image kubecontainer.ImageSpec, pullSecrets []v1.Secr
 
 	userConfigDir, err := ioutil.TempDir("", "rktnetes-user-config-dir-")
 	if err != nil {
-		return fmt.Errorf("rkt: Cannot create a temporary user-config directory: %v", err)
+		return "", fmt.Errorf("rkt: Cannot create a temporary user-config directory: %v", err)
 	}
 	defer os.RemoveAll(userConfigDir)
 
 	config := *r.config
 	config.UserConfigDir = userConfigDir
 
 	if err := r.writeDockerAuthConfig(img, creds, userConfigDir); err != nil {
-		return err
+		return "", err
 	}
 
 	// Today, `--no-store` will fetch the remote image regardless of whether the content of the image
 	// has changed or not. This causes performance downgrades when the image tag is ':latest' and
 	// the image pull policy is 'always'. The issue is tracked in https://github.com/coreos/rkt/issues/2937.
 	if _, err := r.cli.RunCommand(&config, "fetch", "--no-store", dockerPrefix+img); err != nil {
 		glog.Errorf("Failed to fetch: %v", err)
-		return err
+		return "", err
 	}
-	return nil
+	return r.getImageID(img)
 }
 
-func (r *Runtime) IsImagePresent(image kubecontainer.ImageSpec) (bool, error) {
+func (r *Runtime) IsImagePresent(image kubecontainer.ImageSpec) (string, error) {
```

Comment:
It's a little different with `getImageID`, IsImagePresent expects no error if image not found.